src/utils/market-positions-utils.ts

```diff
@@ -56,32 +54,86 @@ function updateNetPositionAndSave(position: MarketPosition): void {
   //       to another address in order to sell them.
   if (position.netQuantity.lt(bigZero)) {
     log.error(
-      'Invalid position: user {} has negative netQuantity on outcome {} on market {}',
-      [position.user, position.outcomeIndex.toString(), position.market],
+      'Invalid position: user {} has negative netQuantity on market {}',
+      [position.user, position.market],
     );
   }
 
   position.save();
 }
 
+/**
+ * Update market position for the maker
+ */
+export function updateMarketPositionFromOrderFilled(
+  maker: string,
+  tokenId: string,
+  side: string,
+  makerAmountFilled: BigInt,
+  takerAmountFilled: BigInt,
+  fee: BigInt,
+): void {
+  // Create/Update market position for the maker
+  let position = getMarketPosition(maker, tokenId);
+  let amountNetFees = takerAmountFilled.minus(fee);
+
+  if (side == TRADE_TYPE_BUY) {
+    position.quantityBought = position.quantityBought.plus(amountNetFees);
+    position.valueBought = position.valueBought.plus(makerAmountFilled);
+  } else {
+    position.quantitySold = position.quantitySold.plus(makerAmountFilled);
+    position.valueSold = position.valueSold.plus(amountNetFees);
+  }
+
+  updateNetPositionAndSave(position);
+}
+
 export function updateMarketPositionFromTrade(event: ethereum.Event): void {
   let transaction = Transaction.load(event.transaction.hash.toHexString());
   if (transaction == null) {
     log.error('Could not find a transaction with hash: {}', [
       event.transaction.hash.toString(),
     ]);
-    throw new Error('Could not find transaction with hash');
+    throw new Error(
+      `Could not find transaction with hash: ${event.transaction.hash.toString()}`,
+    );
   }
 
-  let position = getMarketPosition(
-    transaction.user,
+  const fpmm = FixedProductMarketMaker.load(
     transaction.market,
-    transaction.outcomeIndex,
-  );
+  ) as FixedProductMarketMaker;
 
-  let fee = (FixedProductMarketMaker.load(
-    transaction.market,
-  ) as FixedProductMarketMaker).fee;
+  const conditionalTokenAddress = fpmm.conditionalTokenAddress;
+  const conditions = fpmm.conditions;
+  const collateralToken = fpmm.collateralToken.toString();
+  const outcomeSlotCount = fpmm.outcomeSlotCount as number;
+
+  if (conditions == null) {
+    log.error('LOG: Could not find conditions on the FPMM: {}', [
+      transaction.market,
+    ]);
+    throw new Error(
+      `ERR: Could not find conditions on the FPMM: ${transaction.market.toString()}`,
+    );
+  }
+
+  // Calculate the market/tokenId from the conditionId and the outcome index
+  // Note: this assumes a single conditionId and a zero parentCollectionId
+  let market: string;
+  for (let i = 0; i < conditions.length; i++) {
```

Comment:
if we assume a single conditionId why is the loop necessary?

Reply:
Could have used conditions[0] but the loop seems safer in case the conditions array is empty for some reason